test/AllReduceTests.cpp

```
@@ -242,4 +242,36 @@ namespace RcclUnitTesting
     }
     testBed.Finalize();
   }
+
+  TEST(AllReduce, UserBufferRegistration)
+  {          
+    setenv("UT_PROCESS_MASK", "2", 1);
+    const int nranks = 8;
+    size_t count = 2048;
+    std::vector<int> sendBuff(count, 0);
+    std::vector<int> recvBuff(count, 0);
+    std::vector<int> expected(count, 0);
+
+    for (int i = 0; i < count; ++i){
+        sendBuff[i] = i;
+        expected[i] = i * nranks;
+    }
+    fork_and_launch_rccl(nranks, ncclCollAllReduce, sendBuff, recvBuff, expected);
+  }
+
+  TEST(AllReduce, ManagedMemUserBufferRegistration)
+  {          
+    setenv("UT_PROCESS_MASK", "2", 1);
```

I'm a bit uncomfortable setting the process mask in the test, as this overrides the intentions of the test runner. I think if this test only succeeds for multi-process, we should let it fail for single process and the runner should know not to run it for single process. Or you could check the process mask and skip the test unless it's multi process. Thoughts?

---

I was originally enabling this through an environment variable, however I thought this was not required as it complicates the launching of the tests. I thought that this test should be run as any other, without needing explicit variables. I also thought this could have implications if the CI used it, where we wouldn't want failures. But if the CI executes the test appropriately, I don't see any problem with remove this.

---

Upon further investigation, I think setting this does nothing because this env is parsed at the beginning of the test process and not read again, and the parsed result is only used in TestBed, so setting it here will have no effect.

These should be removed.